api/models/series.go

```diff
@@ -291,7 +291,7 @@ func (series SeriesByTarget) MarshalJSONFast(b []byte) ([]byte, error) {
 		b = append(b, `,"datapoints":[`...)
 		for _, p := range s.Datapoints {
 			b = append(b, '[')
-			if math.IsNaN(p.Val) {
+			if math.IsNaN(p.Val) || math.IsInf(p.Val, 1) || math.IsInf(p.Val, -1) {
```

Comment:
I think this might work as well:
```suggestion
			if math.IsNaN(p.Val) || math.IsInf(p.Val, 0) {
```